packages/block-library/src/navigation-link/hooks.js

```diff
@@ -46,7 +46,11 @@ export function enhanceNavigationLinkVariations( settings, name ) {
 	// Otherwise decorate server passed variations with an icon and isActive function
 	if ( settings.variations ) {
 		const isActive = ( blockAttributes, variationAttributes ) => {
-			return blockAttributes.type === variationAttributes.type;
+			const normalizedVariationKind = variationAttributes.kind.replace(
```

Comment:
I'm not clear why we have two versions of this code...

Reply:
I think the fallback variations will be removed in a later WP version, they're being kept for back compat.

Reply:
The fallback is only used for versions of WP 5.7 and below. Using WP 5.8+ variations are provided by navigation-link in `register_block_type_from_metadata`

https://github.com/WordPress/gutenberg/blob/e65038afdf575eede1b6ebde37afd18e977b1c0d/packages/block-library/src/navigation-link/index.php#L293-L333

We'll be able to remove the entire hook.js file, after [`isActive` supports an array of strings](https://github.com/WordPress/gutenberg/pull/30913) as option, icons map to a string, AND the when the gutenberg plugin moves past <= 5.7 support.